src/mxarray.jl

```diff
@@ -157,13 +158,16 @@ const _mx_is_complex = mxfunc(:mxIsComplex)
 const _mx_is_sparse  = mxfunc(:mxIsSparse)
 const _mx_is_empty   = mxfunc(:mxIsEmpty)
 const _mx_is_struct  = mxfunc(:mxIsStruct)
-const _mx_is_cell    = mxfunc(:mxIsCell) 
+const _mx_is_cell    = mxfunc(:mxIsCell)
 
 
 # getting simple attributes
 
 macro mxget_attr(fun, ret)
-    :( ccall($(fun)::Ptr{Void}, $(ret), (Ptr{Void},), mx.ptr) )
+    quote
+        @assert mx.ptr::Ptr{Void} != C_NULL
```

Comment:
I don't know what other places in the package uses but this shouldn't be a `@assert`, which should only be used for internal consistency check and shouldn't be anything the user can trigger. I usually use `UndefRefError` for this kind of error. This should also use `unsafe_convert` instead but that can be another PR.